anndata/experimental/merge.py

```diff
@@ -0,0 +1,615 @@
+    *,
+    axis: Literal[0, 1] = 0,
+    join: Literal["inner", "outer"] = "inner",
+    merge: Union[StrategiesLiteral, Callable, None] = None,
+    uns_merge: Union[StrategiesLiteral, Callable, None] = None,
+    label: Optional[str] = None,
+    keys: Optional[Collection] = None,
+    index_unique: Optional[str] = None,
+    fill_value: Optional[Any] = None,
+    pairwise: bool = False,
+):
+    """Concatenates multiple AnnData objects along a specified axis using their
+    corresponding stores or paths, and writes the resulting AnnData object
+    to a target location on disk.
+
+    Unlike the `concat` function, this method does not require
+    loading the input AnnData objects into memory,
+    making it a memory-efficient alternative for large datasets.
+    The resulting object written to disk should be equivalent
+    to the concatenation of the loaded AnnData objects using
+    the `concat` function.
+
+    To adjust the maximum amount of data loaded in memory; for sparse
+    arrays use the max_loaded_sparse_elems argument; for dense arrays
+    see the Dask documentation, as the Dask concatenation function is used
+    to concatenate dense arrays in this function
+
+
+
+    Params
+    ------
+    in_files
+        The corresponding stores or paths of AnnData objects to
+        be concatenated. If a Mapping is passed, keys are used for the `keys`
+        argument and values are concatenated.
+    out_file
+        The target path or store to write the result in.
+    overwrite
+        If False the and if a file already exists it will raise an error,
+        otherwise it will overwrite.
+    max_loaded_sparse_elems
+        The maximum number of elements to load in memory when concatenating
+        sparse arrays. Note that this number also includes the empty entries.
+        Set to 100m by default meaning roughly 400mb will be loaded
+        to memory at simultaneously.
+    axis
+        Which axis to concatenate along.
+    join
+        How to align values when concatenating. If "outer", the union of the other axis
+        is taken. If "inner", the intersection. See :doc:`concatenation <../concatenation>`
+        for more.
+    merge
+        How elements not aligned to the axis being concatenated along are selected.
+        Currently implemented strategies include:
+
+        * `None`: No elements are kept.
+        * `"same"`: Elements that are the same in each of the objects.
+        * `"unique"`: Elements for which there is only one possible value.
+        * `"first"`: The first element seen at each from each position.
+        * `"only"`: Elements that show up in only one of the objects.
+    uns_merge
+        How the elements of `.uns` are selected. Uses the same set of strategies as
+        the `merge` argument, except applied recursively.
+    label
+        Column in axis annotation (i.e. `.obs` or `.var`) to place batch information in.
+        If it's None, no column is added.
```

Comment:
Is the “batch” the same thing as the values of the `keys` argument? If so that should be mentioned both in the `in_files` description and the `keys` description.